libr/socket/r2pipe.c

```diff
@@ -233,93 +234,101 @@ R_API R2Pipe *r2pipe_open_dl(const char *libr_path) {
 }
 
 R_API R2Pipe *r2pipe_open(const char *cmd) {
-	R2Pipe *r2pipe = r2pipe_new ();
-	if (!r2pipe) {
+	R2Pipe *r2p = r2pipe_new ();
+	if (!r2p) {
 		return NULL;
 	}
-	if (!cmd) {
-		r2pipe->child = -1;
-		return r2pipe_open_spawn (r2pipe);
+	// r2p->magic = R2P_MAGIC;
+	if (R_STR_ISEMPTY (cmd)) {
+		r2p->child = -1;
+		return r2p_open_spawn (r2p, cmd);
 	}
 #if __WINDOWS__
-	w32_createPipe (r2pipe, cmd);
-	r2pipe->child = (int)(r2pipe->pipe);
+	w32_createPipe (r2p, cmd);
+	r2p->child = (int)(r2p->pipe);
 #else
-	int r = pipe (r2pipe->input);
+	int r = pipe (r2p->input);
 	if (r != 0) {
 		eprintf ("pipe failed on input\n");
-		r2pipe_close (r2pipe);
+		r2pipe_close (r2p);
 		return NULL;
 	}
-	r = pipe (r2pipe->output);
+	r = pipe (r2p->output);
 	if (r != 0) {
 		eprintf ("pipe failed on output\n");
-		r2pipe_close (r2pipe);
+		r2pipe_close (r2p);
 		return NULL;
 	}
 #if LIBC_HAVE_FORK
-	r2pipe->child = fork ();
+	r2p->child = fork ();
 #else
-	r2pipe->child = -1;
+	r2p->child = -1;
 #endif
-	if (r2pipe->child == -1) {
-		r2pipe_close (r2pipe);
+	if (r2p->child == -1) {
+		r2pipe_close (r2p);
 		return NULL;
 	}
-	env ("R2PIPE_IN", r2pipe->input[0]);
-	env ("R2PIPE_OUT", r2pipe->output[1]);
+	env ("R2PIPE_IN", r2p->input[0]);
+	env ("R2PIPE_OUT", r2p->output[1]);
 
-	if (r2pipe->child) {
-		char ch = 1;
+	if (r2p->child) {
+		char ch = -1;
 		// eprintf ("[+] r2pipeipe child is %d\n", r2pipe->child);
-		if (read (r2pipe->output[0], &ch, 1) != 1) {
+		if (read (r2p->output[0], &ch, 1) != 1) {
 			eprintf ("Failed to read 1 byte\n");
-			r2pipe_close (r2pipe);
+			r2pipe_close (r2p);
 			return NULL;
 		}
-		if (ch) {
+		if (ch == -1) {
 			eprintf ("[+] r2pipeipe-io link failed. Expected two null bytes.\n");
```

Comment:
not relevant for merging this PR, though I wonder why here it says it expects 2 null bytes while the read above just read one chars and it doesn't really care about checking nulls.